asio/include/asio/detail/is_buffer_sequence.hpp

```diff
@@ -53,19 +53,22 @@ struct buffer_sequence_memfns_check
 {
 };
 
-template <typename>
-char (&buffer_sequence_begin_helper(...))[2];
-
 #if defined(ASIO_HAS_DECLTYPE)
 
+template <typename>
+char buffer_sequence_begin_helper(...);
+
 template <typename T>
-char buffer_sequence_begin_helper(T* t,
+char (&buffer_sequence_begin_helper(T* t,
     typename enable_if<!is_same<
       decltype(asio::buffer_sequence_begin(*t)),
-        void>::value>::type*);
+        void>::value>::type*))[2];
```

Comment:
why this change?

Reply:
Fallback is now `char` when `ASIO_HAS_DECLTYPE` is defined (see above).
For checker function we need `char[2]`